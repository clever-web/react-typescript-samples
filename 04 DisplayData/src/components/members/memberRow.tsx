import * as React from 'react';
import MemberEntity from '../../api/memberEntity';

interface Props extends React.Props<MemberRow> {
  member : MemberEntity;
}

export default class MemberRow extends React.Component<Props, {}> {

  constructor(props : Props){
        super(props);
  }

  public render() {
      return (
        <tr>
          <td>
            <img src={this.props.member.avatar_url} className="avatar"/>
          </td>
          <td>
            <span>{this.props.member.id}</span>
          </td>
          <td>
            <span>{this.props.member.login}</span>
          </td>
        </tr>
      );
 }
}
