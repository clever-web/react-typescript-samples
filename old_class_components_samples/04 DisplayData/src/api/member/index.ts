import { MemberEntity } from '../../model';
import { members } from './mockData';

const fetchMembers = (): Promise<MemberEntity[]> => {
  return Promise.resolve(members);
};

export const memberAPI = {
  fetchMembers,
};
