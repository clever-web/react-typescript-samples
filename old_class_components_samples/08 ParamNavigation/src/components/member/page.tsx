import * as React from 'react';
import { MemberEntity, MemberErrors } from '../../model';
import { MemberForm } from './memberForm';

interface Props {
  member: MemberEntity;
  memberErrors: MemberErrors;
  onChange: (fieldName: string, value: string) => void;
  onSave: () => void;
}

export const MemberPage: React.StatelessComponent<Props> = (props) => {
  return (
    <MemberForm
      member={props.member}
      memberErrors={props.memberErrors}
      onChange={props.onChange}
      onSave={props.onSave}
    />
  );
}
