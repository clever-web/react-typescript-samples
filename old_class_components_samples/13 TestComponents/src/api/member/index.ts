import { MemberEntity } from '../../model';
import { members } from './mockData';

const baseURL = 'https://api.github.com/orgs/lemoncode';
const userURL = 'https://api.github.com/user';
let mockMembers = members;

const fetchMembers = (): Promise<MemberEntity[]> => {
  return Promise.resolve(mockMembers);
};

const fetchMembersAsync = (): Promise<MemberEntity[]> => {
  const membersURL = `${baseURL}/members`;

  return fetch(membersURL)
    .then((response) => (response.json()))
    .then(mapToMembers);
};

const mapToMembers = (githubMembers: any[]): MemberEntity[] => {
  return githubMembers.map(mapToMember);
};

const mapToMember = (githubMember): MemberEntity => {
  return {
    id: githubMember.id,
    login: githubMember.login,
    avatar_url: githubMember.avatar_url,
  };
};

const saveMember = (member: MemberEntity): Promise<boolean> => {
  const index = mockMembers.findIndex(m => m.id === member.id);

  index >= 0 ?
    updateMember(member, index) :
    insertMember(member);

  return Promise.resolve(true);
};

const updateMember = (member: MemberEntity, index: number) => {
  mockMembers = [
    ...mockMembers.slice(0, index),
    member,
    ...mockMembers.slice(index + 1),
  ];
};

const insertMember = (member: MemberEntity) => {
  member.id = mockMembers.length;

  mockMembers = [
    ...mockMembers,
    member,
  ];
};

const fetchMemberById = (id: number): Promise<MemberEntity> => {
  const member = mockMembers.find(m => m.id === id);

  return Promise.resolve(member);
}

const fetchMemberByIdAsync = (id: number): Promise<MemberEntity> => {
  const membersURL = `${userURL}/${id}`;
  return fetch(membersURL)
    .then((response) => (response.json()))
    .then(mapToMember);
}

export const memberAPI = {
  fetchMembers,
  fetchMembersAsync,
  saveMember,
  fetchMemberById,
  fetchMemberByIdAsync,
  
};
