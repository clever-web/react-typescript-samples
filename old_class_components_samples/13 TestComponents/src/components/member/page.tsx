import * as React from 'react';
import { MemberEntity, MemberErrors } from '../../model';
import { MemberForm } from './memberForm';

export interface Props {
  memberId: number;
  member: MemberEntity;
  memberErrors: MemberErrors;
  fetchMemberById: (id: number) => void;
  onChange: (member: MemberEntity, fieldName: string, value: string) => void;
  onSave: (member: MemberEntity) => void;
}

export class MemberPage extends React.Component<Props, {}> {
  constructor(props) {
    super(props);

    this.onChange = this.onChange.bind(this);
    this.onSave = this.onSave.bind(this);
  }

  componentDidMount() {
    this.props.fetchMemberById(this.props.memberId);
  }

  private onChange(fieldName: string, value: string) {
    this.props.onChange(this.props.member, fieldName, value);
  }

  private onSave() {
    this.props.onSave(this.props.member);
  }

  render() {
    return (
      <MemberForm
        member={this.props.member}
        memberErrors={this.props.memberErrors}
        onChange={this.onChange}
        onSave={this.onSave}
      />
    );
  }
}
