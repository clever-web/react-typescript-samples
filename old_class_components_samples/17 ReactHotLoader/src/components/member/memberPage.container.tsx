import { connect } from 'react-redux';
import { loadMember } from '../../actions/loadMember';
import { uiInputMember } from '../../actions/uiInputMember';
import MemberEntity from '../../api/memberEntity';
import { saveMember } from '../../actions/saveMember';
import { resetSaveCompleted } from '../../actions/resetSaveCompleted';
import { initializeNewMember } from '../../actions/initializeNewMember';
import MemberPage from './memberPage';

let mapStateToProps = (state) => {
    return {
      member: state.member.member
      ,errors : state.member.errors
      ,saveCompleted : state.member.saveCompleted
    }
}

let mapDispatchToProps = (dispatch) => {
  return {
    loadMember: (id : number) => {return dispatch(loadMember(id))}
    ,fireValidationFieldValueChanged: (fieldName : string, value : any) => {return dispatch(uiInputMember(fieldName, value))}
    ,saveMember: (member: MemberEntity) =>  {return dispatch(saveMember(member))}
    ,resetSaveCompletedFlag: () => {return dispatch(resetSaveCompleted())}
    ,initializeNewMember: () => {return dispatch(initializeNewMember())
    }
  }
}

let ContainerMemberPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(MemberPage)


export default ContainerMemberPage;
