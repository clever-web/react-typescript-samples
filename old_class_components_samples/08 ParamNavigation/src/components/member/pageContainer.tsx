import * as React from 'react';
import * as toastr from 'toastr';
import { FieldValidationResult } from 'lc-form-validation';
import { memberAPI } from '../../api/member';
import { MemberEntity, MemberErrors } from '../../model';
import { memberFormValidation } from './memberFormValidation';
import { MemberPage } from './page';

interface Props {
  params: { id: string };
}

interface State {
  member: MemberEntity;
  memberErrors: MemberErrors;
}

export class MemberPageContainer extends React.Component<any, State> {
  constructor(props) {
    super(props);

    this.state = {
      member: {
        id: -1,
        login: '',
        avatar_url: '',
      },
      memberErrors: {
        login: new FieldValidationResult(),
      }
    };

    this.onFieldValueChange = this.onFieldValueChange.bind(this);
    this.onSave = this.onSave.bind(this);
  }

  public componentDidMount() {
    const memberId = Number(this.props.match.params.id) || 0;
    memberAPI.fetchMemberById(memberId)
      .then((member) => {
        this.setState({
          ...this.state,
          member,
        });
      });
  }

  private onFieldValueChange(fieldName: string, value: string) {
    memberFormValidation.validateField(this.state.member, fieldName, value)
      .then((fieldValidationResult) => {
        const nextState = {
          ...this.state,
          member: {
            ...this.state.member,
            [fieldName]: value,
          },
          memberErrors: {
            ...this.state.memberErrors,
            [fieldName]: fieldValidationResult,
          }
        };

        this.setState(nextState);
      });
  }

  private onSave() {
    memberFormValidation.validateForm(this.state.member)
      .then((formValidationResult) => {
        if (formValidationResult.succeeded) {
          memberAPI.saveMember(this.state.member)
            .then(() => {
              toastr.success('Member saved.');
              this.props.history.goBack();
            });
        }
      });
  }

  render() {
    return (
      <MemberPage
        member={this.state.member}
        memberErrors={this.state.memberErrors}
        onChange={this.onFieldValueChange}
        onSave={this.onSave}
      />
    );
  }
}
