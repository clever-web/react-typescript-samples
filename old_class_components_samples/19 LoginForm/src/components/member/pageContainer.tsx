import * as React from 'react';
import * as toastr from 'toastr';
import { RouteComponentProps } from 'react-router';
import { FieldValidationResult } from 'lc-form-validation';
import { memberAPI } from '../../api/member';
import { MemberEntity, MemberErrors } from '../../model';
import { memberFormValidation } from './memberFormValidation';
import { MemberPage } from './page';
import { Header } from '../header';

interface Params {
  id: string;
}

interface Props extends RouteComponentProps<Params> {
}

export const MemberPageContainer: React.StatelessComponent<Props> = (props: Props) => {

  const [member, setMember] = React.useState({
    id: -1,
    login: '',
    avatar_url: '',
  });

  const [memberErrors, setMemberErrors] = React.useState({
    login: new FieldValidationResult(),
  });

  const loadMember = () => {
    const memberId = Number(props.match.params.id) || 0;
    memberAPI.fetchMemberById(memberId)
      .then((member) => setMember(member));
  }

  React.useEffect(() => {
    loadMember();
  }, []);

  const onFieldValueChange = (fieldName: string, value: string) => {
    memberFormValidation.validateField(member, fieldName, value)
      .then((fieldValidationResult) => {
        setMember({
          ...member,
          [fieldName]: value
        });
        setMemberErrors({
          ...memberErrors,
          [fieldName]: fieldValidationResult,
        });
      });
  }

  const onSave = () => {
    memberFormValidation.validateForm(member)
      .then((formValidationResult) => {
        if (formValidationResult.succeeded) {
          memberAPI.saveMember(member)
            .then(() => {
              toastr.success('Member saved.');
              props.history.goBack();
            });
        }
      });
  }

  return (
    <div>
      <Header />
      <MemberPage
        member={member}
        memberErrors={memberErrors}
        onChange={onFieldValueChange}
        onSave={onSave}
      />
    </div>
  );
}
