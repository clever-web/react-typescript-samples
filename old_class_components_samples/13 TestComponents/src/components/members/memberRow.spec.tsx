import * as React from 'react';
import {MemberRow,Props} from './memberRow';
import {shallow} from 'enzyme';
import { createEmptyMember } from '../../reducers/member';

describe('/components/members/memberRow',()=>{
    it('should render as expected whenprops are passed',()=>{
        //arrange    
        let member: Props = {member:createEmptyMember()};
        //act
            let component = shallow(    
                <MemberRow{...member}/>
            );
        //assert
        expect (component).toMatchSnapshot();
    });
})

