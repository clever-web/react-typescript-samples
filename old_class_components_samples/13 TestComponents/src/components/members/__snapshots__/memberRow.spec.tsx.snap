// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/members/memberRow should render as expected whenprops are passed 1`] = `
<tr>
  <td>
    <img
      className="avatar"
      src=""
    />
  </td>
  <td>
    <Link
      replace={false}
      to="/member/-1"
    >
      -1
    </Link>
  </td>
  <td>
    <span />
  </td>
</tr>
`;
